src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('values is less than 1');
+    }
 
-export class Circle {
+    const longestSide = Math.max(this.a, this.b, this.c);
+    const perimetr = this.a + this.b + this.c;
 
+    if (longestSide >= perimetr - longestSide) {
+      throw new Error(
+        'longest side of a triangle is bigger than a sum of two others',
+      );
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
```

Comment:
```suggestion
    const halfP = (this.a + this.b + this.c) / 2;
```
in fact it is half perimeter

Reply:
ok, but [p](https://ru.wikipedia.org/wiki/%D0%A4%D0%BE%D1%80%D0%BC%D1%83%D0%BB%D0%B0_%D0%93%D0%B5%D1%80%D0%BE%D0%BD%D0%B0) it`s mean a halp of perimetr

Reply:
I just consider more meaningful variable for code.

> ok, but [p]